### DISCUSSION TITLE
egl.rs is create_pbuffer_surface needed? (wasm32-unknown-emscripten)

### DISCUSSION THREAD

**Discussion by caiiiycuk** [gfx-rs/wgpu#2225] [General]
Hi, I am trying to add support for wasm32-unknown-emscirpten platform. Seems that everything goes well, however egl.rs making call to create_pbuffer_surface (eglCreatePbufferSurface). This function is not supported by emscripten, all other used functions are ok. So, for now, it's only blocker to build wgpu with emscripten, is it possible to avoid using this function? I saw in code that comments about switching context, maybe it can be avoided in browser environment?

**Comment by kvark** [comment]
It *is* avoided in the browser when building for `wasm32-unknown-unknown`. Do you need `wasm32-unknown-emscripten` specifically? Our EGL usage is a little tricky, I'm not surprised WASM target doesn't like it.

**Reply by kvark** [reply]
The pbuffer there isn't really used. It's only for being able to make EGL context current without any actual surface in it. I have a feeling that the whole concept of "make EGL current" works in a very limited way on the Web anyway. So try to see what happens if `pbuffer: None` on this path.

**Reply by caiiiycuk** [reply]
Yeah I want to use emscripten toolkit with wgpu, because it's more powerful then wasm32-unknown-unknow and fits match better for porting. Also I think there is no real blockers to add wasm32-unknow-emscipten support. So I want to test it, and if it's ok, then create a PR.
